lfucg-exactions/server/notes/viewsets.py

```diff
@@ -23,39 +23,62 @@ def get_queryset(self):
-
-                if parent_content_type_string is not None:
-                    split_parent_string = parent_content_type_string.split('_')
-
-                    if len(split_parent_string) == 2:
-                        parent_content_type_app_label = split_parent_string[0]
-                        parent_content_type_model = split_parent_string[1]
-
-                        parent_content_type = ContentType.objects.get(app_label=parent_content_type_app_label, model=parent_content_type_model)
-
-                        if parent_content_type and child_content_type:
-                            query_list = queryset.filter(
-                                Q(content_type=parent_content_type, object_id=parent_object_id) |
-                                Q(content_type=child_content_type, object_id=child_object_id))
-
-                        queryset = query_list
-
+                parent_content_type = None
+                parent_object_id = None
+                grandparent_content_type = None
+                grandparent_object_id = None
+                if child_content_type_model == 'lot':
+                    content_lot = Lot.objects.filter(id=child_object_id)
+                    if content_lot.count() > 0 and content_lot[0].plat:
+                        chosen_lot = content_lot[0]
+                        content_plat = Plat.objects.filter(id=chosen_lot.plat.id)
+                        chosen_plat = content_plat[0]
+                        parent_object_id = chosen_plat.id
+                        parent_content_type = ContentType.objects.get_for_model(Plat)
+                        if chosen_plat and chosen_plat.subdivision:
+                            grandparent_content_type = ContentType.objects.get_for_model(Subdivision)
+                            grandparent_object_id = chosen_plat.subdivision.id
+                        else:
+                            grandparent_content_type = None
+                            grandparent_object_id = None
+                    else:
+                        parent_content_type = None
+                        parent_object_id = None
+                        grandparent_content_type = None
+                        grandparent_object_id = None
+                elif child_content_type_model == 'plat':
+                    content_plat = Plat.objects.filter(id=child_object_id)
+                    if content_plat.count() > 0 and content_plat[0].subdivision:
+                        chosen_plat = content_plat[0]
+                        parent_content_type = ContentType.objects.get_for_model(Subdivision)
+                        parent_object_id = chosen_plat.subdivision.id
+                    else:
+                        parent_content_type = None
+                        parent_object_id = None
+
+                if grandparent_content_type is not None and parent_content_type is not None and child_content_type:
```

Comment:
you can't have a grandparent without a parent and you can't have a parent without a child_content_type. So you can just check if grandparent is not none, elif parent is not none, else.